source/features/search-results-filter.tsx

```diff
@@ -1,95 +1,66 @@
 import {React} from 'dom-chef/react';
 import select from 'select-dom';
 import features from '../libs/features';
-import domify from '../libs/domify';
 
-function cleanLinks(links) {
-	for (const link of links) {
+function cleanLinks() {
+	for (const link of select.all('.menu-item')) {
 		const searchParams = new URLSearchParams(link.search);
 		searchParams.set('q', cleanSearchQuery(searchParams.get('q')));
 		link.search = searchParams.toString();
 	}
 }
 
 function getSearchQuery() {
-	return new URL(location.href).searchParams.get('q');
+	return new URLSearchParams(location.search).get('q');
 }
 
-function cleanSearchQuery(query) {
-	return query.replace(/\s*\bis:(pr|issue)\b\s*/gi, '');
+function cleanSearchQuery(query, newType = '') {
+	return query
+		.replace(/\bis:(pr|issue)\b/gi, newType)
+		.replace(/\s{2,}/g, ' ');
 }
 
-function createUrl(type) {
-	const url = new URL(location.pathname, location.origin); // Create url without any search params.
-	url.searchParams.set('q', cleanSearchQuery(getSearchQuery()) + ` is:${type}`);
+function createUrl(type, pathname = location.pathname) {
+	const url = new URL(pathname, location.origin);
+	url.searchParams.set('q', cleanSearchQuery(getSearchQuery(), `is:${type}`));
```

Comment:
This assumes that `is:type` is in the query, but that is not the case when freshly opening the search page.